digits/dataset/generic/test_views.py

```diff
@@ -0,0 +1,251 @@
+        url = '/datasets/generic/create/%s' % cls.EXTENSION_ID
+        if request_json:
+            url += '.json'
+
+        rv = cls.app.post(url, data=data)
+
+        if request_json:
+            if rv.status_code != 200:
+                raise RuntimeError('Model creation failed with %s' % rv.status_code)
+            return json.loads(rv.data)['id']
+
+        # expect a redirect
+        if not 300 <= rv.status_code <= 310:
+            s = BeautifulSoup(rv.data, 'html.parser')
+            div = s.select('div.alert-danger')
+            if div:
+                print div[0]
+            else:
+                print rv.data
+            raise RuntimeError('Failed to create dataset - status %s' % rv.status_code)
+
+        job_id = cls.job_id_from_response(rv)
+
+        assert cls.dataset_exists(job_id), 'dataset not found after successful creation'
+
+        cls.created_datasets.append(job_id)
+        return job_id
+
+    @classmethod
+    def setUpClass(cls, **kwargs):
+        super(BaseViewsTestWithDataset, cls).setUpClass()
+        cls.dataset_id = cls.create_dataset(json=True, **kwargs)
+        assert cls.dataset_wait_completion(cls.dataset_id) == 'Done', 'create failed'
+
+class GenericViewsTest(BaseViewsTest):
+    def test_page_dataset_new(self):
+        rv = self.app.get('/datasets/generic/new/%s' % self.EXTENSION_ID)
+        assert rv.status_code == 200, 'page load failed with %s' % rv.status_code
+        assert extensions.data.get_extension(self.EXTENSION_ID).get_title() in rv.data, 'unexpected page format'
+
+    def test_nonexistent_dataset(self):
+        assert not self.dataset_exists('foo'), "dataset shouldn't exist"
+
+class GenericCreationTest(BaseViewsTestWithDataset):
+    """
+    Dataset creation tests
+    """
+
+    def test_create_json(self):
+        job_id = self.create_dataset(json=True)
+        self.abort_dataset(job_id)
+
+    def test_create_delete(self):
+        job_id = self.create_dataset()
+        assert self.delete_dataset(job_id) == 200, 'delete failed'
+        assert not self.dataset_exists(job_id), 'dataset exists after delete'
+
+    def test_create_abort_delete(self):
+        job_id = self.create_dataset()
+        assert self.abort_dataset(job_id) == 200, 'abort failed'
+        assert self.delete_dataset(job_id) == 200, 'delete failed'
+        assert not self.dataset_exists(job_id), 'dataset exists after delete'
+
+    def test_create_wait_delete(self):
+        job_id = self.create_dataset()
+        assert self.dataset_wait_completion(job_id) == 'Done', 'create failed'
+        assert self.delete_dataset(job_id) == 200, 'delete failed'
+        assert not self.dataset_exists(job_id), 'dataset exists after delete'
+
+    def test_invalid_number_of_reader_threads(self):
+        try:
+            job_id = self.create_dataset(json=True, dsopts_num_threads=0)
+            assert self.dataset_wait_completion(job_id) != 'Done', 'should have failed'
+        except RuntimeError:
```

Comment:
Your checks seem weird. Do you mean to allow the job to fail in one of two ways (either status != 'Done' or `RuntimeError`)?